packages/xo-server/src/xapi/mixins/networking.js

```diff
@@ -24,8 +25,28 @@ export default {
     await this._disconnectVif(this.getObject(vifId))
   },
   editVif: makeEditObject({
-    ipv4Allowed: true,
-    ipv6Allowed: true,
+    ipv4Allowed: {
+      get: true,
+      set: [
+        'ipv4Allowed',
+        function (value, vif) {
+          if (!isEmpty(value) && vif.locking_mode !== 'locked') {
```

Comment:
`value.length > 0`

Reply:
A suggestion is easier to apply and non-null is enough :slightly_smiling_face: 

```suggestion
          if (value.length !== 0 && vif.locking_mode !== 'locked') {
```